query/src/pipelines/transforms/group_by/aggregator_polymorphic_keys.rs

```diff
@@ -94,123 +96,111 @@ pub trait PolymorphicKeysHelper<Method: HashMethod> {
     ) -> Self::GroupColumnsBuilder;
 }
 
-impl PolymorphicKeysHelper<HashMethodKeysU8> for HashMethodKeysU8 {
-    type State = ShortFixedKeysAggregatorState<u8>;
-    fn aggregate_state(&self) -> Self::State {
-        Self::State::create((u8::MAX as usize) + 1)
-    }
+macro_rules! dispatch_unsigned_short_types{
```

Comment:
Why use macros? It doesn't seem to reduce the duplicate code for us. Instead, it reduces the code readability(and macros are not modification friendly).